core/src/main/java/org/apache/iceberg/SchemaUpdate.java

```diff
@@ -61,6 +61,7 @@ class SchemaUpdate implements UpdateSchema {
   private int lastColumnId;
   private boolean allowIncompatibleChanges = false;
   private Set<String> identifierFieldNames;
+  private boolean caseSensitive;
```

Comment:
Can we make the initialization explicit here, I believe we want this to default to "true" since that was the previous behavior.

Reply:
done.